src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,214 @@
+            ImmutableArray.Create(PreferStreamAsyncMemoryOverloads.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (!(root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node))
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (!(model.GetOperation(node, ct) is IInvocationOperation invocation))
+            {
+                return;
+            }
+
+            // Defensive check to ensure the fix is only attempted on one of the 4 specific undesired overloads
+            if (invocation.Arguments.Length < 3 || invocation.Arguments.Length > 4)
+            {
+                return;
+            }
+
+            IArgumentOperation? bufferOperation = GetArgumentByPositionOrName(invocation.Arguments, 0, "buffer", out bool isBufferNamed);
+            if (bufferOperation == null)
+            {
+                return;
+            }
+
+            IArgumentOperation? offsetOperation = GetArgumentByPositionOrName(invocation.Arguments, 1, "offset", out bool isOffsetNamed);
+            if (offsetOperation == null)
+            {
+                return;
+            }
+
+            IArgumentOperation? countOperation = GetArgumentByPositionOrName(invocation.Arguments, 2, "count", out bool isCountNamed);
+            if (countOperation == null)
+            {
+                return;
+            }
+
+            // No nullcheck for this, because there is an overload that may not contain it
+            IArgumentOperation? cancellationTokenOperation = GetArgumentByPositionOrName(invocation.Arguments, 3, "cancellationToken", out bool isCancellationTokenNamed);
+
+            string title = MicrosoftNetCoreAnalyzersResources.PreferStreamAsyncMemoryOverloadsTitle;
+
+            Task<Document> fixInvocation = FixInvocation(doc, root, invocation, invocation.TargetMethod.Name,
+                                                         bufferOperation.Value.Syntax, isBufferNamed,
+                                                         offsetOperation.Value.Syntax, isOffsetNamed,
+                                                         countOperation.Value.Syntax, isCountNamed,
+                                                         cancellationTokenOperation?.Value.Syntax, isCancellationTokenNamed);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: title,
+                    createChangedDocument: c => fixInvocation,
+                    equivalenceKey: title),
```

Comment:
Append method name to title to ensure that FixAll operation only fixes violations with matching method name.

```suggestion
                    equivalenceKey: title + invocation.TargetMethod.Name),
```

Reply:
@mavasani do I need to do this for both the title and the equivalenceKey?

Reply:
Okay so I appended the method name to both, and the tests passed. Let me know if this was the wrong thing to do so I only apply it to the equivalenceKey as in your code suggestion.

Reply:
Only to the equivalence key. Title is the user facing string that is shown to users in light bulb, and we want that to be the localized string without method name.

Reply:
As suggested below, I should only do this for equivalenceKey, not for title. Reverted.